bindings/trycp_runner/src/common.rs

```diff
@@ -45,6 +56,253 @@ pub fn connect_trycp_client<SV: UserValuesConstraint>(
+///     install_app(ctx, Path::new("path/to/your/happ").to_path_buf(), &"your_role_name".to_string())?;
+///     try_wait_for_min_peers(ctx, Duration::from_secs(60))?;
+///     Ok(())
+/// }
+/// ```
+///
+/// Note that if no apps have been installed, you are waiting for too many peers, or anything else
+/// prevents enough peers being discovered then the function will wait up to the `wait_for` duration
+/// before continuing. It will not fail if too few peers were discovered.
+///
+/// Note that the smallest resolution is 1s. This is because the function will sleep between
+/// querying peers from the conductor. You could probably not use this function for performance
+/// testing peer discovery!
+pub fn try_wait_for_min_peers<SV>(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<SV>>,
+    wait_for: Duration,
+) -> HookResult
+where
+    SV: UserValuesConstraint,
+{
+    static MIN_PEERS: OnceLock<usize> = OnceLock::new();
+
+    let client = ctx.get().trycp_client();
+    let agent_id = ctx.agent_id().to_string();
+
+    let min_peers = *MIN_PEERS.get_or_init(|| {
+        std::env::var("MIN_PEERS")
+            .ok()
+            .map(|s| s.parse().expect("MIN_PEERS must be a number"))
+            .unwrap_or(2)
+    });
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            let start_discovery = Instant::now();
+            for _ in 0..wait_for.as_secs() {
+                let agent_list = client.agent_info(agent_id.clone(), None, None).await?;
+
+                if agent_list.len() >= min_peers {
+                    break;
+                }
+
+                tokio::time::sleep(Duration::from_secs(1)).await;
+            }
+
+            println!(
+                "Discovery for agent {} took: {}s",
+                agent_id,
+                start_discovery.elapsed().as_secs()
+            );
+
+            Ok(())
+        })?;
+
+    Ok(())
+}
+
+/// Shuts down the Holochain conductor managed by the TryCP server for the current agent.
+///
+/// You *MUST* call this function in your agent teardown. Otherwise, dropping the agent context will
+/// panic when the TryCP client is dropped.
+///
+/// ```rust
+/// use trycp_wind_tunnel_runner::prelude::{TryCPRunnerContext, AgentContext, TryCPAgentContext, HookResult, shutdown_remote};
+///
+/// fn agent_teardown(ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext>) -> HookResult {
+///     shutdown_remote(ctx)?;
+///     Ok(())
+/// }
+/// ```
+pub fn shutdown_remote<SV>(
+    ctx: &mut AgentContext<TryCPRunnerContext, TryCPAgentContext<SV>>,
+) -> HookResult
+where
+    SV: UserValuesConstraint,
+{
+    let client = ctx.get().trycp_client();
+    let agent_id = ctx.agent_id().to_string();
+
+    ctx.runner_context()
+        .executor()
+        .execute_in_place(async move {
+            client.shutdown(agent_id.clone(), None, None).await?;
```

Comment:
I know we run reset as part of the startup, but do we want to run it as part of shutdown too (maybe with `let _ = `) so we don't leave the databases of previous tests laying around?

Reply:
Yep, particularly for tests that create more data than this one does, that's probably a good pattern to put in place.

Added this to the agent tear down and handled the error at that level